### PR TITLE
feat(views): a more flexible extendable html page shell

### PR DESCRIPTION
**Pull request by hypeJunction**
Makes default and admin page shells more flexible by providing additional
extension points, as well as allowing more parameters to be passed to
underlying views

### CODE REVIEW

File: views/default/page/default.php [hunk 1]
```diff
@@ -1,58 +1,43 @@
...
+	'topbar' => elgg_view('page/elements/topbar', $vars),
+	'header' => elgg_view('page/elements/header', $vars),
+	'navbar' => elgg_view('page/elements/navbar', $vars),
+	'body' => elgg_view('page/elements/body', $vars),
+	'footer' => elgg_view('page/elements/footer', $vars),
+];
+
+$page = '';
+foreach ($sections as $section => $content) {
+	$page .= elgg_view("page/elements/$section/before", $vars);
```
Comment (mrclay) [inline views/default/page/default.php]: Check existence or it'll complain in logs

Comment (mrclay) [inline views/default/page/default.php]: Maybe the before/after logic could be rolled into the section view?

Comment (hypeJunction) [inline views/default/page/default.php]: I suppose we can drop it. If really wanted, you can extend `page/elements/section` and sniff the section name before outputting content.

**Comment by hypeJunction** [comment]
@mrclay Addressed your comment

**Review by mrclay — APPROVED** [review]

**Comment by mrclay** [comment]
Would be nice to get another 👍

**Pull request merged**